### ISSUE TITLE
iptables dns redirection for hsts

### ISSUE THREAD

**Issue by edawnpipe933** [byt3bl33d3r/MITMf#232]
In the current version the iptables rule for dns redirection will only be set if enabling dns spoofing (--spoof --dns). If only using arp spoofing and sslstrip with hsts bypass (--spoof --arp --hsts) without dns spoofing, the dns requests won't be redirected to the local dns server. This causes on the victims site that the modified urls for hsts bypass can not be loaded (connection error).

Possible solution:
Changing plugins/spoof.py line 71 from
```python
if options.dns:
```
to
```python
if options.dns or options.hsts:
```

**Comment by brunoaduarte** [comment]
Exactly... there's no point using --hsts (domain redirect) without the "--spoof --dns" flags, cause it would never work

**Comment by brandontict** [comment]
Making the above changes did not work in my case. 

I have always ran into the issue " DNS Probe did not respond in time " or along the lines of something similar to that error . 

Thanks for the tip :+1:

**Comment by byt3bl33d3r** [comment]
@edawnpipe933 will look into this, thanks!

**Comment by edawnpipe933** [comment]
@brunoaduarte:
You are right that for hsts bypassing dns spoofing respectively domain redirection has to be done. That's part of the magic. But it is possible to start mitmf without dns spoofing flags and only using "--spoof --arp --hsts". And this combination causes the problems mentioned by me above. Sure one could deny this combination. But from my point of view it is totally fine to make the dns spoofing implicit to the hsts flag. However this requires the right iptables to be set like mentioned in my post.
This solution is tested by me and runs perfectly good.

@brandontict:
Did you make the changes I mentioned in my first post?
Are your iptables set correctly (for dns spoofing too) when mitmf is running?
Do you despite your error see a redirection to the faked pages (e.g. www**w** instead of www) at your victim?
I tested this solution on two different clients with different victims and it worked pretty fine. Without my fix the victim got redirected to the faked url but couldn't load it obviously (conn. error). If you got a different error, maybe there is an other problem too.

@byt3bl33d3r:
Thank you very much! Great work on a great project.
